go-controller/hybrid-overlay/pkg/controller/node_linux_test.go

```diff
@@ -33,45 +30,99 @@ import (
 )
 
 const (
-	testOVSMAC     string = "11:22:33:44:55:66"
-	testMgmtMAC    string = "06:05:04:03:02:01"
-	testDRMAC      string = "00:00:00:7a:af:04"
-	testNodeSubnet string = "1.2.3.0/24"
-	testNodeIP     string = "1.2.3.3"
+	testMgmtMAC string = "06:05:04:03:02:01"
+
+	thisNodeSubnet string = "1.2.3.0/24"
+	thisNodeDRIP   string = "1.2.3.3"
+	thisNodeDRMAC  string = "22:33:44:55:66:77"
 )
 
+// returns if the two flowCaches are the same
+func compareFlowCache(returnedFlowCache, expectedFlowCache map[string]*flowCacheEntry) bool {
+	if len(returnedFlowCache) != len(expectedFlowCache) {
+		return false
+	}
+
+	for key, entry := range returnedFlowCache {
+		expectedEntry, ok := expectedFlowCache[key]
+		if !ok {
+			klog.Errorf("Unexpected entry %s in nodes flowCache", entry)
+			return false
+		}
+		if !compareFlowCacheEntry(entry, expectedEntry) {
+			klog.Errorf("The test nodes flowCache entry  %s contains unexpected inforation %s", key, entry)
+			return false
+		}
+	}
+	return true
+
+}
+
+// compares two entries in the flow cache
+func compareFlowCacheEntry(returnedEntry, expectedEntry *flowCacheEntry) bool {
+	if returnedEntry.learnedFlow != expectedEntry.learnedFlow ||
+		returnedEntry.ignoreLearn != expectedEntry.ignoreLearn ||
+		len(returnedEntry.flows) != len(expectedEntry.flows) {
+		return false
+	}
+
+	for key, returnedEntryFlow := range returnedEntry.flows {
+		if returnedEntryFlow != expectedEntry.flows[key] {
+			return false
+		}
+
+	}
+
+	return true
+}
+
+func generateInitialFlowCacheEntry(mgmtInterfaceAddr string) *flowCacheEntry {
+	mgmtPortLink, err := netlink.LinkByName(types.K8sMgmtIntfName)
+	Expect(err).NotTo(HaveOccurred())
+	mgmtPortMAC := mgmtPortLink.Attrs().HardwareAddr
```

Comment:
@JacobTanenbaum could we put this bit into BeforeEach(), and then just pass the hardware address into this function?

Reply:
I don't think so, the problem is that the netns stuff does not actually get executed until the `appRun()` is executed so `mgmtPortLink, err := netlink.LinkByName(types.K8sMgmtIntfName)` will return an error if run from `BeforeEach()` because the Link is not actually set up